include/zephyr/net/mqtt_sn.h

```diff
@@ -397,6 +397,17 @@ int mqtt_sn_publish(struct mqtt_sn_client *client, enum mqtt_sn_qos qos,
  */
 int mqtt_sn_input(struct mqtt_sn_client *client);
 
+/**
+ * @brief Get topic name by topic ID
+ *
+ * @param[in] client The MQTT-SN client that uses this topic.
+ * @param[in] id Topic identifier.
+ * @param[out] len Pointer to name string length or NULL if not used.
```

Comment:
Shall `len` really be optional? Pity it's not documented in `struct mqtt_sn_topic`, but I don't think topic name can be treated as a NULL terminated string. And if that's the case, the acual data length shall always be provided with the data pointer (name).

Reply:
You are right, topic name is not guarantied to be terminated, so `len` should be mandatory.

Though for the sake of usage simplicity I would prefer to store it terminated. On the other hand, looks like everywhere in mqtt and mqtt-sn libs strings are not terminated.

Reply:
Changed the API to match better with the rest of the library